### ISSUE TITLE
ClassCastException when using eventbus sockjs bridge

### ISSUE THREAD

**Issue by blueberrynotblue** [vert-x3/vertx-lang-scala#65]
My verticle:
```
class HttpVerticle extends ScalaVerticle {
  val logger = Logger.getLogger(this.getClass.getSimpleName)

  override def startFuture(): Future[_] = {
    val router = Router.router(vertx)
    val sockJSHandler = SockJSHandler.create(vertx).bridge(
      BridgeOptions()
        .setInboundPermitted(mutable.Buffer(PermittedOptions().setAddress("testAddress")))
        .setOutboundPermitted(mutable.Buffer(PermittedOptions().setAddress("testAddress")))
    )
    router.route("/eventbus/*").handler(sockJSHandler)
    vertx
      .createHttpServer()
      .requestHandler(router.accept _)
      .listenFuture(8666, "0.0.0.0")
  }
}
```
In html head:
```
<script src="https://cdn.jsdelivr.net/npm/sockjs-client@1/dist/sockjs.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/vertx/3.5.1/vertx-eventbus.min.js"></script>
```
My js script:
```
<script>
    var eb=new EventBus("http://****/eventbus")
    eb.onopen=function () {
        console.log("eventbus opened~~~")
        // send(address,body,header,handler)
        eb.send("testAddress",{document:"some_doc"},{action:"dont move"},function (reply) {
            console.log(reply);
        })
    }
</script>
```
Eventbus was succesfully opened. **eb.send** will cause exception on backend:
```
java.lang.ClassCastException: scala.collection.immutable.Map$Map1 cannot be cast to io.vertx.core.json.JsonObject
        at io.vertx.core.json.JsonObject.getJsonObject(JsonObject.java:216)
        at io.vertx.ext.web.handler.sockjs.impl.EventBusBridgeImpl.doSendOrPub(EventBusBridgeImpl.java:374)
        at io.vertx.ext.web.handler.sockjs.impl.EventBusBridgeImpl.lambda$internalHandleSendOrPub$5(EventBusBridgeImpl.java:194)
        at io.vertx.ext.web.handler.sockjs.impl.EventBusBridgeImpl.checkCallHook(EventBusBridgeImpl.java:159)
        at io.vertx.ext.web.handler.sockjs.impl.EventBusBridgeImpl.internalHandleSendOrPub(EventBusBridgeImpl.java:187)
        at io.vertx.ext.web.handler.sockjs.impl.EventBusBridgeImpl.handleSocketData(EventBusBridgeImpl.java:137)
        at io.vertx.ext.web.handler.sockjs.impl.EventBusBridgeImpl.lambda$null$18(EventBusBridgeImpl.java:314)
        at io.vertx.ext.web.handler.sockjs.impl.SockJSSession.handleMessages(SockJSSession.java:372)
        at io.vertx.ext.web.handler.sockjs.impl.WebSocketTransport$WebSocketListener.handleMessages(WebSocketTransport.java:113)
        at io.vertx.core.http.impl.WebSocketImplBase$FrameAggregator.handleTextFrame(WebSocketImplBase.java:334)
        at io.vertx.core.http.impl.WebSocketImplBase$FrameAggregator.handle(WebSocketImplBase.java:300)
        at io.vertx.core.http.impl.WebSocketImplBase$FrameAggregator.handle(WebSocketImplBase.java:289)
        at io.vertx.core.http.impl.WebSocketImplBase.handleFrame(WebSocketImplBase.java:282)
        at io.vertx.core.http.impl.Http1xServerConnection.handleOther(Http1xServerConnection.java:474)
        at io.vertx.core.http.impl.Http1xServerConnection.processMessage(Http1xServerConnection.java:439)
        at io.vertx.core.http.impl.Http1xServerConnection.handleMessage(Http1xServerConnection.java:141)
        at io.vertx.core.http.impl.HttpServerImpl$ServerHandlerWithWebSockets.handleMessage(HttpServerImpl.java:693)
        at io.vertx.core.http.impl.HttpServerImpl$ServerHandlerWithWebSockets.handleMessage(HttpServerImpl.java:636)
        at io.vertx.core.net.impl.VertxHandler.lambda$channelRead$1(VertxHandler.java:146)
        at io.vertx.core.impl.ContextImpl.lambda$wrapTask$2(ContextImpl.java:337)
        at io.vertx.core.impl.ContextImpl.executeFromIO(ContextImpl.java:195)
        at io.vertx.core.net.impl.VertxHandler.channelRead(VertxHandler.java:144)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:310)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:284)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1359)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:935)
        at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:141)
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:645)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:580)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:497)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:459)
        at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:886)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.lang.Thread.run(Thread.java:748)
```

**Comment by blueberrynotblue** [comment]
change HttpVerticle into java verticle, then it works fine.

**Comment by codepitbull** [comment]
This looks like to have the same cause as #66. If so it is a regression introduced with how I handle the refgistration of custom Scala-serializers with the new Jackson version in 3.5.1.

**Comment by codepitbull** [comment]
Fix will be part of 3.5.2

**Comment by blueberrynotblue** [comment]
@codepitbull Hi~
 When cluster member removed, exception occured.
![Image](http://m.qpic.cn/psb?/63a6f161-f590-44a9-909d-a6040fc82d3e/0ml5PcpwSKfPBjzTHUaLswtzCwTK0GySQqj7u9eZM8k!/b/dEMBAAAAAAAA&bo=1wQGBAAAAAADB*M!&rf=viewer_4) 
Is this same cause as above?
